src/python/grpcio_tests/tests/interop/xds_interop_client.py

```diff
@@ -0,0 +1,201 @@
+
+    def await_rpc_stats_response(self, timeout_sec: int
+                                ) -> messages_pb2.LoadBalancerStatsResponse:
+        """Blocks until a full response has been collected."""
+        with self._lock:
+            self._condition.wait_for(lambda: not self._rpcs_needed,
+                                     timeout=float(timeout_sec))
+            response = messages_pb2.LoadBalancerStatsResponse()
+            for peer, count in self._rpcs_by_peer.items():
+                response.rpcs_by_peer[peer] = count
+            response.num_failures = self._no_remote_peer + self._rpcs_needed
+        return response
+
+
+_global_lock = threading.Lock()
+_stop_event = threading.Event()
+_global_rpc_id: int = 0
+_watchers: Set[_StatsWatcher] = set()
+_global_server = None
+
+
+def _handle_sigint(sig, frame):
+    _stop_event.set()
+    _global_server.stop(None)
+
+
+class _LoadBalancerStatsServicer(test_pb2_grpc.LoadBalancerStatsServiceServicer
+                                ):
+
+    def __init__(self):
+        super(_LoadBalancerStatsServicer).__init__()
+
+    def GetClientStats(self, request: messages_pb2.LoadBalancerStatsRequest,
+                       context: grpc.ServicerContext
+                      ) -> messages_pb2.LoadBalancerStatsResponse:
+        print("Received stats request.")
+        sys.stdout.flush()
+        start = None
+        end = None
+        watcher = None
+        with _global_lock:
+            start = _global_rpc_id + 1
+            end = start + request.num_rpcs
+            watcher = _StatsWatcher(start, end)
+            _watchers.add(watcher)
+        response = watcher.await_rpc_stats_response(request.timeout_sec)
+        with _global_lock:
+            _watchers.remove(watcher)
+        return response
+
+
+# TODO: Accept finer-grained arguments.
+def _run_single_channel(args: argparse.Namespace):
+    global _global_rpc_id  # pylint: disable=global-statement
+    duration_per_query = 1.0 / float(args.qps)
+    with grpc.insecure_channel(args.server) as channel:
+        stub = test_pb2_grpc.TestServiceStub(channel)
+        while not _stop_event.is_set():
+            request_id = None
+            with _global_lock:
+                request_id = _global_rpc_id
+                _global_rpc_id += 1
+            print(f"Sending request to backend: {request_id}")
+            sys.stdout.flush()
+            start = time.time()
+            end = start + duration_per_query
+            call, _ = stub.UnaryCall.with_call(messages_pb2.SimpleRequest(),
+                                               timeout=float(
+                                                   args.rpc_timeout_sec))
+            print(f"Got result {request_id}")
+            sys.stdout.flush()
+            with _global_lock:
+                for watcher in _watchers:
+                    # TODO: Implement a peer details getter.
+                    peer = f"192.168.1.{request_id % 255}"
```

Comment:
The peer for the `LoadBalancerStatsResponse.rpcs_by_peer` should actually come from the server's response (so `SimpleResponse.hostname`, as done [here](https://github.com/grpc/grpc/blob/5312a3aeb2c06e6d8c518365e4a701456bd653c3/test/cpp/interop/xds_interop_client.cc#L142)). Getting the peer from the context (ala `context.peer()` in the C++ client) is only used if `-print_response` is true, which is only used by an internal test that should soon be deprecated, so I don't think there's any need for Python to implement this flag or pull the peer from the context at all.